btk/survey.py

```diff
@@ -56,90 +65,91 @@
+        survey = galcheat.get_survey(survey_name)
+        filters = []
+        for band, filtr in survey.get_filters().items():
+            if psf_func is None:
+                psf = get_default_galcheat_psf(survey, filtr)
+            else:
+                psf = psf_func(survey, filtr)
+            btk_filter = Filter(
+                band,
+                psf,
+                filtr.sky_brightness.value,
+                filtr.exposure_time.value,
+                filtr.zeropoint.value,
+                filtr.extinction.value,
+            )
+            filters.append(btk_filter)
+
+        btk_survey = Survey(
+            survey.name,
+            survey.pixel_scale.value,
+            survey.effective_area.value,
+            survey.mirror_diameter.value,
+            survey.airmass.value,
+            survey.zeropoint_airmass.value,
+            filters,
+        )
+        btk_surveys.append(btk_survey)
 
+    if len(btk_surveys) == 1:
+        return btk_surveys[0]
+    return btk_surveys
 
-def get_surveys(names="Rubin", overrides: Iterable = ()):
-    """Return specified surveys as `btk.survey.Survey` objects.
 
-    NOTE: The surveys currently implemented correspond to config files inside `conf/surveys`. See
-    the documentation for how to add your own surveys via custom config files.
+def get_default_galcheat_psf(survey: galcheat.survey.Survey, filtr: galcheat.filter.Filter):
+    """Return the default PSF model as a galsim object based on galcheat survey parameters.
 
     Args:
-        names (str or list): A single str specifying a survey from conf/surveys or a list with
-            multiple survey names.
-        overrides (Iterable): List or tuple containg overrides for the survey config files. An
-            example element of overrides could be 'surveys.Rubin.airmass=1.1', i.e. what you would
-            pass into the CLI in order to customize the surveys used (here specified by `names`).
+        survey (galcheat.survey.Survey): Survey object from galcheat.
+        filtr (galcheat.filter.Filter): Filter object from galcheat.
 
     Returns:
         btk.survey.Survey object or list of such objects.
     """
-    if isinstance(names, str):
-        names = [names]
-    if not isinstance(names, list):
-        raise TypeError("Argument 'names' of `get_surveys` should be a str or list.")
-    overrides = [f"surveys={names}", *overrides]
-    surveys = []
-    with initialize(config_path="../conf"):
-        cfg = compose("config", overrides=overrides)
-    for survey_name in cfg.surveys:
-        survey_conf = cfg.surveys[survey_name]
-        surveys.append(get_survey_from_cfg(survey_conf))
-    if len(surveys) == 1:
-        return surveys[0]
-    return surveys
-
-
-def get_psf(
+    filt_wavelength = _central_wavelenghts[filtr.name]
+    return get_default_psf(
+        survey.mirror_diameter.value,
+        survey.effective_area.value,
+        filtr.psf_fwhm.value,
+        filt_wavelength=filt_wavelength,
```

Comment:
Here when you will use the `effective_wavelength` from galcheat, make sure you convert it to angstroms first.

Reply:
I thought we wanted nanometers see documentation below? (which is the default in galcheat I think?)

Reply:
See this [line](https://github.com/LSSTDESC/BlendingToolKit/blob/cf3db28baa2d7d8ddba16ef503533dafb278b69b/btk/survey.py#L178). Actually I'm confused why this is 1e-10 and not 1e-9. Do you know @thuiop ?

Reply:
The wavelengths in _central_wavelength were in 1e-10 meters for some reason (not sure why, this comes from WLD).

Reply:
Ah I forgot the Angstrom definition, so that is why there is a 1e-10. Ok I fixed it